packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -0,0 +1,56 @@
+
+ /**
+ * This adds {fileName, lineNumber} annotations to React component definitions
+ * and to jsx tag literals.
+ *
+ *
+ * == JSX Literals ==
+ *
+ * <sometag />
+ *
+ * becomes:
+ *
+ * <sometag __source={{fileName: 'this/file.js', lineNumber: 10}}/>
```

Comment:
React Native will do what ever this done. I'm just worried that DEV mode is becoming unusable. I've already heard stories of people turning it off during development to iterate quicker. We need to regain some of that and keep perf regressions to a minimum.